examples/summarization/bart/evaluate_cnn.py

```diff
@@ -19,42 +20,83 @@ def chunks(lst, n):
 def generate_summaries(
     examples: list, out_file: str, model_name: str, batch_size: int = 8, device: str = DEFAULT_DEVICE
 ):
-    fout = Path(out_file).open("w")
-    model = BartForConditionalGeneration.from_pretrained(model_name).to(device)
-    tokenizer = BartTokenizer.from_pretrained("facebook/bart-large")
+    fout = Path(out_file).open("w", encoding="utf-8")
+    if "t5" in model_name:
+        model = T5ForConditionalGeneration.from_pretrained(model_name).to(device)
+        tokenizer = T5Tokenizer.from_pretrained(model_name)
+        # update config with summarization specific params
+        task_specific_params = model.config.task_specific_params
+        if task_specific_params is not None:
+            model.config.update(task_specific_params.get("summarization", {}))
```

Comment:
do we want this for bart also?

Can we use AutoModel/AutoTokenizer?
facebook/bart-large-cnn tokenizer is identical

Reply:
I have added `task_specific_params` for bart also.
Tried using AutoModel but throws errors with regards to architecture for both T5/Bart.
AutoTokenizer seems to work seamlessly!

Reply:
Try `AutoModelWithLMHead` and send traceback if it breaks?

Reply:
`AutoModelWithLMHead` seems to work seamlessly!